engine/src/main/java/com/google/android/fhir/sync/CurrentSyncJobStatus.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.fhir.sync
+
+import java.time.OffsetDateTime
+
+/**
+ * Sealed class representing different states of a synchronization operation. It combines
+ * [WorkInfo.State] and [SyncJobStatus]. Enqueued state represents [WorkInfo.State.ENQUEUED] where
+ * [SyncJobStatus] is not applicable. Running state is a combined state of [WorkInfo.State.ENQUEUED]
+ * and [SyncJobStatus.Started] or [SyncJobStatus.InProgress]. Succeeded state is a combined state of
+ * [WorkInfo.State.SUCCEEDED] and [SyncJobStatus.Started] or [SyncJobStatus.Succeeded]. Failed state
+ * is a combined state of [WorkInfo.State.FAILED] and [SyncJobStatus.Failed]. Cancelled state
+ * represents [WorkInfo.State.CANCELLED] where [SyncJobStatus] is not applicable.
+ */
+sealed class CurrentSyncJobStatus {
```

Comment:
For periodic, it is called PeriodicSyncState, not Status. let us be consistent
```suggestion
sealed class CurrentSyncJobState {
```

Reply:
done.